src/test/java/uk/gov/hmcts/probate/service/CaseQueryServiceTest.java

```diff
@@ -154,4 +155,12 @@ public void findCasesForGrantAwaitingDocs() {
         assertThat(cases.get(0).getId(), is(1L));
         assertEquals("Smith", cases.get(0).getData().getDeceasedSurname());
     }
+
+    @Test
+    public void testExceptionWithNullFromRestTemplatePost() {
+        when(restTemplate.postForObject(any(), any(), any())).thenReturn(null);
+
+        Assertions.assertThatThrownBy(() -> caseQueryService.findCasesWithDatedDocument("testDate"))
+            .isInstanceOf(ClientDataException.class);
+    }
```

Comment:
why dont you use the same test structure here as you do with the FeeServiceTest below?